src/test/kotlin/org/jmailen/gradle/kotlinter/functional/CustomTaskTest.kt

```diff
@@ -0,0 +1,198 @@
+        projectRoot.resolve(".editorconfig") {
+            writeText(editorConfig)
+        }
+        projectRoot.resolve("build.gradle") {
+            @Language("groovy")
+            val buildScript = """
+                import org.jmailen.gradle.kotlinter.tasks.LintTask
+    
+                task customizedLintTask(type: LintTask) {
+                    source files('src')
+                    reports = ['plain': file('build/lint-report.txt')]
+                    indentSize = 123
+                    continuationIndentSize = 17
+                    experimentalRules = true
+                    disabledRules = ["final-newline"]
+                    editorConfigPath = project.rootProject.file(".editorconfig")
+                    ignoreFailures = true
+                    fileBatchSize = 12
+                }
+                
+            """.trimIndent()
+            appendText(buildScript)
+        }
+
+        build("customizedLintTask").apply {
+            assertEquals(TaskOutcome.SUCCESS, task(":customizedLintTask")?.outcome)
+        }
+    }
+
+    @Test
+    fun `ktLintFormat custom task succeeds with fully provided configuration`() {
+        projectRoot.resolve(".editorconfig") {
+            writeText(editorConfig)
+        }
+        projectRoot.resolve("build.gradle") {
+            @Language("groovy")
+            val buildScript = """
+                import org.jmailen.gradle.kotlinter.tasks.FormatTask
+    
+                task customizedFormatTask(type: FormatTask) {
+                    source files('src')
+                    indentSize = 123
+                    continuationIndentSize = 17
+                    experimentalRules = true
+                    disabledRules = ["final-newline"]
+                    editorConfigPath = project.rootProject.file(".editorconfig")
+                    fileBatchSize = 12
+                }
+                
+            """.trimIndent()
+            appendText(buildScript)
+        }
+
+        build("customizedFormatTask").apply {
+            assertEquals(TaskOutcome.SUCCESS, task(":customizedFormatTask")?.outcome)
+        }
+    }
+
+    @Test
+    fun `ktLint custom task skips reports generation if reports not passed`() {
+        projectRoot.resolve("src/main/kotlin/MissingNewLine.kt") {
+            @Language("kotlin")
+            val validClass = """
+                class MissingNewLine
+            """.trimIndent()
+            writeText(validClass)
+        }
+        projectRoot.resolve("build.gradle") {
+            @Language("groovy")
+            val buildScript = """
+                import org.jmailen.gradle.kotlinter.tasks.LintTask
+    
+                task reportsNotConfigured(type: LintTask) {
+                    source files('src')
+                }
+                
+                task reportsEmpty(type: LintTask) {
+                    source files('src')
+                    reports = [:]
+                }
```

Comment:
FYI: I added test covering current behavior. I'll update them once we agree what should be a valid behavior for an empty map.